low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -22,28 +22,233 @@
+                        "resource_id": f"{sas_rid}:{rid.strip()}",
+                        "negotiated_link_rate": rate.strip()
+                    })
+                    if "Gbit".lower() in rate.strip().lower():
+                        health.append("OK")
+
+            # By some logic set health status, description, recommendation
+            health_data.append(health)
+            health_status = "OK" if "OK" in health else "Fault"
+            self.set_health_data(r_data, health_status)
+            sas_hba_data.append(r_data)
+
+        if leaf_node == "sas_hba":
+            return sas_hba_data
+
+        sas_ports_data = []
+        for heath_list in health_data:
+            for i, health in enumerate(heath_list):
+                if i % 4:
+                    continue
+
+                sas_port_data = self.get_data_template(f'sas_port_{i/4+1}',
+                                                       False)
+                sas_port_data["health"]["specifics"].append({
+                    "negotiated_link_rate": "12.0 Gbit"
+                })
+
+                health_status = "OK" if "OK" in heath_list[i:i+4] else "Fault"
+                self.set_health_data(sas_port_data, health_status)
+                sas_ports_data.append(sas_port_data)
+
+        return sas_ports_data
+
+    def get_nw_ports_info(self):
+        """Return the Network ports information."""
+        network_cable_data = []
+        io_counters = psutil.net_io_counters(pernic=True)
+
+        for interface, addrs in psutil.net_if_addrs().items():
+            nic_info = self.get_data_template(interface, False)
+
+            specifics = {}
+            for addr in addrs:
+                if addr.family == socket.AF_INET:
+                    specifics["ipV4"] = addr.address
+
+            if interface in io_counters:
+                io_info = io_counters[interface]
+                specifics.update({
+                    "networkErrors": io_info.errin + io_info.errout,
+                    "droppedPacketsIn": io_info.dropin,
+                    "droppedPacketsOut": io_info.dropout,
+                    "packetsIn": io_info.packets_recv,
+                    "packetsOut": io_info.packets_sent,
+                    "trafficIn": io_info.bytes_recv,
+                    "trafficOut": io_info.bytes_sent
+                })
+
+            nw_status, nw_cable_conn_status = \
+                self.get_nw_status(interface)
+
+            specifics["nwStatus"] = nw_status
+            specifics["nwCableConnStatus"] = nw_cable_conn_status
+            nic_info["health"]["specifics"].append(specifics)
+
+            # set health status, description and recommendation
+            if nw_status == "UP" and nw_cable_conn_status == "OK":
+                health_status = "OK"
+            elif nw_status == "UNKNOWN" and nw_cable_conn_status == "NA":
+                health_status = "NA"
+            else:
+                health_status = "Fault"
```

Comment:
Tried to derive some logic to state the health, doesn't look good to me, please suggest some logic if possible.